File: src/CSharp/DynamicBuffers.CSharp/FieldNamePattern.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DynamicBuffers;

/// <summary>
/// Specifies field name matching logic.
/// </summary>
public enum FieldNamePattern
{
    /// <summary> Checks only CSharp naming style. </summary>
    CSharpName,

    /// <summary> Checks only JSON naming style. </summary>
    JsonName,

    /// <summary> Checks only .proto default name. </summary>
    ProtoName,
}

File: src/Python/dynamicbuffers/field_name_pattern.py
#!/usr/bin/env python3
"""
Field name pattern
"""
from enum import Enum


class FieldNamePattern(Enum):
    """
    Specifies field name matching logic.
    """

    PythonName = 0  # noqa: E221
    """ Checks only Python naming style. """

    JsonName   = 1  # noqa: E221
    """ Checks only JSON naming style. """

    ProtoName  = 2  # noqa: E221
    """ Checks only .proto default name. """
